### PR TITLE
chore: fixes compile error

### LINKED ISSUE #13: Error Running cargo install for M2 Air
The command: `cargo install --git https://github.com/matter-labs/era-test-node.git` produces the following output (after installing Rust with `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh` and adding to PATH with `source "$HOME/.cargo/env"`):

```bash
cargo install --git https://github.com/matter-labs/era-test-node.git
    Updating git repository `https://github.com/matter-labs/era-test-node.git`
  Installing era_test_node v1.0.1 (https://github.com/matter-labs/era-test-node.git#fbda966a)
    Updating crates.io index
    Updating git repository `https://github.com/matter-labs/jsonrpc.git`
    Updating git repository `https://github.com/matter-labs/zksync-era.git`
    Updating git repository `https://github.com/matter-labs/era-zk_evm.git`
    Updating git repository `https://github.com/matter-labs/era-zkEVM-assembly.git`
    Updating git repository `https://github.com/matter-labs/solidity_plonk_verifier.git`
    Updating git repository `https://github.com/matter-labs/era-zkevm_test_harness.git`
    Updating git repository `https://github.com/matter-labs/era-zkevm_opcode_defs.git`
    Updating git repository `https://github.com/yoshidan/google-cloud-rust`
error: failed to compile `era_test_node v1.0.1 (https://github.com/matter-labs/era-test-node.git#fbda966a)`, intermediate artifacts can be found at `/var/folders/xk/93skcyt518l_b5sgpktx98vc0000gn/T/cargo-install2RZDR9`

Caused by:
  no matching package named `google-cloud-default` found
  location searched: https://github.com/yoshidan/google-cloud-rust?branch=main
  required by package `zksync_object_store v1.0.0 (https://github.com/matter-labs/zksync-era.git?rev=6c91b1a6aec0552f7a5a6a78d6ddac07d43aa4bc#6c91b1a6)`
      ... which satisfies git dependency `zksync_object_store` of package `zksync_core v1.0.0 (https://github.com/matter-labs/zksync-era.git?rev=6c91b1a6aec0552f7a5a6a78d6ddac07d43aa4bc#6c91b1a6)`
      ... which satisfies git dependency `zksync_core` of package `era_test_node v1.0.1 (/Users/dennis/.cargo/git/checkouts/era-test-node-0a2946bd389429a8/fbda966)`
```

System specs:

```bash
Software:

    System Software Overview:

      System Version: macOS 13.4.1 (22F82)
      Kernel Version: Darwin 22.5.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 1 day, 23 hours, 5 minutes

Hardware:

    Hardware Overview:

      Model Name: MacBook Air
      Model Identifier: Mac14,2
      Chip: Apple M2
      Total Number of Cores: 8 (4 performance and 4 efficiency)
      Memory: 24 GB
      System Firmware Version: 8422.121.1
```

### LINKED ISSUE #14: Error Installing test-node with Mac (Intel CPU)
When I run the cargo install command I'm getting a compile error stating that the google-cloud-default package can not be found at 

https://github.com/yoshidan/google-cloud-rust?branch=main

This is a dependency in the zksync_object_store package found in the zksync-era/core/lib repo. I'm currently trying to fix the issue locally by changing the dependency to the crate.io google-cloud-default created by yoshida. This fixes the dependency problem, but there are other errors thrown from the ClientConfig methods not being found in the gcs.rs file in the object_store's src directory. I think maybe the git google-cloud-rust fork has undergone some breaking changes and no longer works with the zksync implementation. I could be misunderstanding the issue though so some help and clarity on this would be greatly appreciated.

### PR DESCRIPTION
**Pull request by dutterbutter**
# What :computer: 
* Update dependencies to latest on zksync-era
* Updates node.rs and zk.rs to return a BoxFuture type from the jsonrpc_core library

# Why :hand:
* To reflect the latest changes in zksync-era and fix errors with compiling and running cargo install

# Evidence :camera:

Run: `make rust-build` 

Screenshot 2023-07-17 at 10 03 46 PM

# Notes :memo:
Closes #13 Closes #14

### CODE REVIEW

File: src/fork.rs [hunk 1]
```diff
@@ -61,7 +76,7 @@ impl ForkStorage {
                 raw_storage: InMemoryStorage::with_system_contracts_and_chain_id(
                     chain_id,
                     hash_bytecode,
-                    dev_use_local_contracts,
```
Comment (MexicanAce) [inline src/fork.rs:64]: We need to keep this parameter to maintain functionality of using local bootloader/system contracts

File: src/node.rs [hunk 2]
```diff
@@ -238,8 +246,8 @@ impl InMemoryNode {
         let mut inner = self.inner.write().unwrap();
         let keys = {
             let mut storage_view = StorageView::new(&inner.fork_storage);
-            storage_view.set_value(&key, u256_to_h256(U256::from(10u128.pow(22))));
-            storage_view.get_modified_storage_keys().clone()
+            storage_view.set_value(key, u256_to_h256(U256::from(10u128.pow(19))));
```
Comment (MexicanAce) [inline src/node.rs:249]: Did you intend to reduce the amount of ETH we give Rich accounts?

Comment (dutterbutter) [reply src/node.rs:249]: Nope will update to reflect the previous change.

Comment (dutterbutter) [reply src/node.rs]: @MexicanAce updated.

File: src/node.rs [hunk 3]
```diff
@@ -252,17 +260,18 @@ impl InMemoryNode {
         let execution_mode = TxExecutionMode::EthCall {
             missed_storage_invocation_limit: 1000000,
         };
+        let (mut block_context, block_properties) = create_test_block_params();
```
Comment (mm-zk) [inline src/node.rs:263]: use create_block_context() (as it sets stuff like l1_gas_price corretctly -- that create_test_block_params doesn't)

**Comment by dutterbutter** [comment]
@mm-zk updated with requested changes.

**Review by mm-zk — APPROVED** [review]

**Pull request merged**